libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/NavigationTelemetry.kt

```diff
@@ -158,29 +158,23 @@ internal object NavigationTelemetry : NavigationMetricListener {
      * @param directionsRoute new route passed to [MapboxNavigation]
      */
     fun updateSessionRoute(directionsRoute: DirectionsRoute) {
-        val navigationBuilder = navigationSessionState.toBuilder()
-            .tripIdentifier(TelemetryUtils.obtainUniversalUniqueIdentifier())
-        navigationBuilder.currentDirectionRoute(directionsRoute)
-        eventDispatcher.addMetricEventListeners(this)
-
-        if (isOffRoute) {
-            // If we are off-route, update the reroute count
-            navigationBuilder.rerouteCount(navigationSessionState.rerouteCount() + 1)
-            val hasRouteOptions = directionsRoute.routeOptions() != null
-            navigationBuilder.requestIdentifier(
-                if (hasRouteOptions) {
-                    directionsRoute.routeOptions()?.requestUuid()
-                } else {
-                    null
-                }
-            )
-            navigationSessionState = navigationBuilder.build()
-            updateLastRerouteEvent(directionsRoute)
-            lastRerouteDate = Date()
-            isOffRoute = false
-        } else {
-            // Not current off-route - update the session
-            navigationSessionState = navigationBuilder.build()
+        navigationSessionState.let {
```

Comment:
Should we add a parameter `sessionState ->` so it's clear what `it` means afterwards? Would that be more readable?